File: src/main/java/de/steklopod/sec03/Lec03FluxTake.java
package de.steklopod.sec03;

import de.steklopod.utils.Util;
import reactor.core.publisher.Flux;

public class Lec03FluxTake {

    public static void main(String[] args) {

        // map
        // filter
        Flux.range(1, 10)
                .log()
                .take(3) // cancels
                .log()
                .subscribe(Util.subscriber());


    }

}

File: src/main/kotlin/de/steklopod/sec03/Lec03FluxTake.kt
package de.steklopod.sec03

import de.steklopod.utils.Util.subscriber
import reactor.core.publisher.Flux

object Lec03FluxTake {
    @JvmStatic
    fun main(args: Array<String>) {

        // map
        // filter
        Flux.range(1, 10)
            .log()
            .take(3) // cancels
            .log()
            .subscribe(subscriber())
    }
}
